File: app/src/main/java/com/example/task2/BenchmarksApp.java
package com.example.task2;

import android.app.Application;

import com.example.task2.models.BenchmarkComponent;
import com.example.task2.models.DaggerBenchmarkComponent;

public class BenchmarksApp extends Application {
    private static BenchmarksApp instance;
    private BenchmarkComponent component;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        component = DaggerBenchmarkComponent.create();
    }

    public static BenchmarksApp getInstance() {
        return instance;
    }

    public BenchmarkComponent getComponent() {
        return component;
    }
}

File: app/src/main/java/com/example/task2/BenchmarksApp.kt
package com.example.task2

import android.app.Application
import com.example.task2.models.BenchmarkComponent
import com.example.task2.models.DaggerBenchmarkComponent
import com.example.task2.models.DaggerRoomComponent
import com.example.task2.models.RoomComponent
import com.example.task2.models.RoomModule

class BenchmarksApp : Application() {

    val roomComponent: RoomComponent?
        get() = Companion.roomComponent

    val component: BenchmarkComponent?
        get() = Companion.component

    companion object {
        @JvmStatic
        var instance: BenchmarksApp? = null
            private set
        private var component: BenchmarkComponent? = null
        private var roomComponent: RoomComponent? = null

        @JvmStatic
        fun setAppComponent(testComponent: BenchmarkComponent?) {
            component = testComponent
        }
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
        Companion.component = DaggerBenchmarkComponent.create()
        Companion.roomComponent = DaggerRoomComponent.builder()
                .roomModule(RoomModule(applicationContext))
                .build()
    }
}
